p2p/libp2p/netMessenger.go

```diff
@@ -613,32 +629,55 @@ func (netMes *networkMessenger) SetPeerShardResolver(peerShardResolver p2p.PeerS
 		return p2p.ErrNilPeerShardResolver
 	}
 
-	kadSharder, err := networksharding.NewKadSharder(prioBits, peerShardResolver)
+	reconnecter, _ := netMes.peerDiscoverer.(p2p.Reconnecter)
```

Comment:
treat second variable if !ok - make a nil reconnecter or return ErrWrongTypeAssertion.

Reply:
yes, sharderFactory.Create does not accept a nil reconnecter. It will reutrn an error on the create() method.

Reply:
If the reconnecter is nil, the factory will return a nil connection monitor implementation. Added TODO to refactor in a future PR this area.